modules/ank/arrow-ank/src/main/kotlin/arrow/ank/algebra.kt

```diff
@@ -1,33 +1,34 @@
 package arrow.ank
 
+import arrow.Kind
+import arrow.core.Option
+import arrow.core.Try
 import arrow.core.Tuple2
+import arrow.data.Nel
+import arrow.typeclasses.MonadThrow
 import org.intellij.markdown.ast.ASTNode
-import java.io.File
+import java.nio.file.Path
 
-fun createTarget(source: File, target: File): File =
-  createTargetImpl(source, target)
+interface AnkOps<F> {
+  fun MF(): MonadThrow<F>
 
-fun getFileCandidates(target: File): List<File> =
-  getFileCandidatesImpl(target)
+  fun createTargetDirectory(source: Path, target: Path): Kind<F, Try<Path>>
```

Comment:
There is no point on having concrete `Kind<F, Try<Path>` return type since they can just be `Kind<F, Path>` given you already have in scope a constrain on `MonadThrow<F>`. Why is the internal `Try` needed? You may just call `MF().raiseError` or use `MF().catch { ... }` if you just need to control synchronous error control. We can make the entire set of ops return `IO<A>` if you'd rather use an explicit concrete data type.

Reply:
Yeah, I kinda forgot when defining the algebra and then did not change it when implementing ...
Will do in a sec. Just removing a bunch of them should be fine since `MonadDefer.delay` also catches errors.
I had the type as `Kind<F, A>` because I was starting with only Monad as the constraint on F. That would make testing with something like an id interpreter nice. Now with MonadThrow this may not be valid anymore. (Although Try might be a good fit there)